### DISCUSSION TITLE
How to document responses that poem does automatically?

### DISCUSSION THREAD

**Discussion by hffmnn** [poem-web/poem#521] [Q&A]
Hi everyone,

given I have a function defined like this

```
async fn create(&self, todo: Json<CreateTodo>) -> Result<CreateResponse>
```

with these possible responses:

```
#[derive(ApiResponse)]
pub enum CreateResponse {
    #[oai(status = 201)]
    Success(Json<Todo>),

    #[oai(status = 404)]
    NotFound,
}
```

How would I let `poem-openapi` know that there are other possible errors, like a `400 - Bad Request` that gets handled automatically by poem (which is great)? Would I have to add it manually to every possible `ApiResponse`? If so: how would I do that?

And: What about other things like panic. When adding the `CatchPanic` middleware I get a `500 - Internal Server Error` (again: this is great), but how to document it?

**Comment by banool** [comment]
Personally I have "solved" this by just making variants for those responses that I know Poem can return in the response enum. Not ideal, but the best I've got.
